### ISSUE TITLE
API: Add a way to discover file ids from a dataset to enable use in download API.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#1795]
Users have been asking how to download files. We have a method but you need to know the file id.
We need an api method, most likely at the dataset level, to discover the contents, ie. file ids that can be used for download.

**Comment by pdurbin** [comment]
This related ticket has an emphasis on file IDs for **published** datasets: #1717

**Comment by michbarsinai** [comment]
This can be easily added to the get dataset version command - we can list the file metadata there. But - we need to know which files to show to whom.

@scolapasta @eaquigley what is the logic of showing file metadata to a user? Which permissions should I look into, and how does the `isRestricted` method of `DataFile` comes into play?

**Comment by scolapasta** [comment]
@michbarsinai If a user can see a version, they can see all the files associated with that version (via the file metadata objects). isRestricted applies only for download.

**Comment by pdurbin** [comment]
@michbarsinai  over all, it looks good but two things:

1. Below we have camel case such as "contentType" but in the Search API I used "file_content_type". I'm actually thinking the Search API should change. There's so much camelCase other places I should probably embrace it.
2. Today in chat @astrofrog [clued me in](http://irclog.iq.harvard.edu/dataverse/2015-03-31#i_17686) to the fact that we're exposing the path on disk to the files. We don't want this, do we? Example below of "glassfish/domains/domain1..."

curl -s https://apitest.dataverse.org/api/v1/datasets/10/versions?key=$KEY | jq .data[].files

```
[
  {
    "description": "",
    "label": "trees.png",
    "version": 1,
    "datasetVersionId": 1,
    "datafile": {
      "id": 12,
      "name": "trees.png",
      "contentType": "image/png",
      "filename": "%2Fusr%2Flocal%2Fglassfish4%2Fglassfish%2Fdomains%2Fdomain1%2Ffiles%2F10.5072%2FFK2%2FIOOAJF%2F14c7157584b-15f561a376ee",
      "originalFormatLabel": "UNKNOWN",
      "md5": "0386269a5acb2c57b4eade587ff4db64",
      "description": ""
    }
  }
]
```

**Comment by michbarsinai** [comment]
This is created at JsonPrinter line 341. @landreev - any input on why `dataFile.getFilename()` returns the  path?

**Comment by pdurbin** [comment]
> curl -s https://apitest.dataverse.org/api/v1/datasets/10/versions?key=$KEY | jq .data[].files

This API call is nice but the big problem is that users have to know that their dataset has a dataset id of 10 in this case. Instead they should be able to use DOI, which is what #1837 is about.

**Comment by pdurbin** [comment]
#1837 was fixed in Dataverse 4.3.
